cpp/src_prims/linalg/reduce_rows_by_key.h

```diff
@@ -334,18 +347,42 @@ void reduce_rows_by_key(const DataIteratorT d_A, int lda,
     // with doubles we have ~20% speed up - with floats we can hope something around 2x
     // Converting d_keys to char
     convert_array(d_keys_char, d_keys, nrows, stream);
-    sum_rows_by_key_small_nkeys(d_A, lda, d_keys_char, nrows, ncols, nkeys,
-                                d_sums, stream);
+    sum_rows_by_key_small_nkeys(d_A, lda, d_keys_char, d_weights, nrows, ncols,
+                                nkeys, d_sums, stream);
   } else {
     for (KeyType key_offset = 0; key_offset < nkeys;
          key_offset += SUM_ROWS_BY_KEY_LARGE_K_MAX_K) {
       KeyType this_call_nkeys = std::min(SUM_ROWS_BY_KEY_LARGE_K_MAX_K, nkeys);
-      sum_rows_by_key_large_nkeys_rowmajor(d_A, lda, d_keys, nrows, ncols,
-                                           key_offset, this_call_nkeys, d_sums,
-                                           stream);
+      sum_rows_by_key_large_nkeys_rowmajor(d_A, lda, d_keys, d_weights, nrows,
+                                           ncols, key_offset, this_call_nkeys,
+                                           d_sums, stream);
     }
   }
 }
 
+/**
```

Comment:
can you add prims unit-tests for weighted and non-weighted reduce_rows_by_key, please?

Reply:
done